### PR TITLE
Add overlay on mobile to make swipe-to-move discoverable

### PR DESCRIPTION
**Pull request by joshlory**
Request from Alice:

screen shot 2015-11-12 at 2 25 49 pm

Only shows up on mobile (desktop is unchanged).

### CODE REVIEW

**Comment by davidsbailey** [comment]
This looks good in English, but I am concerned that splitting this sentence across two messages will make it hard to translate. For example, the word "then" may naturally come at the very end of the sentence in another language, and translators might not be able to see both strings at the same time in order to move words between the two.

If we can contain the sentence in a single DOM element that would solve the problem. Can we accomplish that by using either a newline part way through the string, or just let the text naturally wrap onto the next line?

If you don't want to spend possibly unnecessary time reimplementing this, we could wait for Tanya to return to get a definitive answer on the translations concern.

**Comment by joshlory** [comment]
Good call on other locales, but the issue we'll run into is that SVG doesn't have any built-in wrapping.  Let's sync with @tanyaparker tomorrow when she's back in the office.

File: apps/src/studio/studio.js [hunk 1]
```diff
@@ -443,14 +447,48 @@ var drawMap = function () {
...
 
-  var resetText = document.createElementNS(SVG_NS, 'text');
-  resetText.setAttribute('id', 'resetText');
-  resetText.setAttribute('class', 'studio-reset-text');
-  resetText.setAttribute('x', Studio.MAZE_WIDTH / 2);
-  resetText.setAttribute('y', RESET_TEXT_Y_POSITION);
-  resetText.appendChild(document.createTextNode(studioMsg.tapOrClickToReset()));
-  resetText.setAttribute('visibility', 'visible');
-  gameTextGroup.appendChild(resetText);
+  if (dom.isMobile()) {
```
Comment (davidsbailey) [inline apps/src/studio/studio.js]: Can you find out whether we want this for Windows Touch too? If so, you can add `dom.isWindowsTouch` to the condition here.

File: apps/src/studio/studio.js [hunk 2]
```diff
@@ -443,14 +447,48 @@ var drawMap = function () {
...
+    resetOverlayRect.setAttribute('width', Studio.MAZE_WIDTH);
+    resetOverlayRect.setAttribute('height', Studio.MAZE_HEIGHT);
+    resetOverlayRect.setAttribute('fill', 'black');
+    resetOverlayRect.setAttribute('opacity', 0.3);
+    overlayGroup.appendChild(resetOverlayRect);
+    var resetTextA = document.createElementNS(SVG_NS, 'text');
+    resetTextA.setAttribute('id', 'resetTextA');
+    resetTextA.setAttribute('class', 'studio-reset-text');
+    resetTextA.setAttribute('x', Studio.MAZE_WIDTH / 2);
+    resetTextA.setAttribute('y', RESET_TEXT_Y_POSITION - 30);
```
Comment (davidsbailey) [inline apps/src/studio/studio.js:459]: The number `30` looks like it's been chosen to make the text line up in English. Let's come up with a solution that works for other languages.

Comment (davidsbailey) [inline apps/src/studio/studio.js:459]: you can disregard this comment, I see that the `30` is not language-specific.

**Comment by davidsbailey** [comment]
Sounds good. It may be possible to split a single translation into two strings of approximately equal length, but that sounds kind of scary and the place we choose to split may look weird in other languages.

@islemaster was dealing with an SVG overlay yesterday and today so he may have some ideas too.

**Comment by islemaster** [comment]
See the epic that is #4253.  If you don't need to click through your overlay though, there may be no issues.

**Comment by tanyaparker** [comment]
Could make them two strings "Tap to play" and next line "Drag to move" as separate suggestions. Still makes sense in English and would be easy to translate.

**Comment by aoby** [comment]
I updated with a larger image from @kozzi, and changed the text to "Tap to play", "Swipe to move". @alicesteinglass 

screen shot 2015-12-01 at 3 43 19 pm

**Comment by aoby** [comment]
Updated the overlay again, resized and repositioned it:

screen shot 2015-12-01 at 4 14 56 pm

**Comment by aoby** [comment]
@davidsbailey, I'm reviving this PR. Do you have any more feedback? Thx

File: apps/src/studio/studio.js [hunk 3]
```diff
@@ -2108,12 +2146,28 @@ Studio.reset = function(first) {
     .setAttribute('visibility', 'hidden');
   document.getElementById('victoryText')
     .setAttribute('visibility', 'hidden');
-  var resetText = document.getElementById('resetText');
-  if (level.tapSvgToRunAndReset) {
-    resetText.textContent = studioMsg.tapOrClickToPlay();
-    resetText.setAttribute('visibility', 'visible');
+  if (dom.isMobile() || dom.isWindowsTouch()) {
```
Comment (davidsbailey) [inline apps/src/studio/studio.js:2149]: This looks like it maight show up in other games which use studio. Can you please verify that this doesn't show up unexpectedly in `/s/playlab/stage/1/puzzle/1` (playlab) or `/s/algebra/stage/7/puzzle/1` (big game)?

Comment (davidsbailey) [inline apps/src/studio/studio.js:2149]: checking those levels on mobile / touch of course

File: apps/src/studio/studio.js [hunk 4]
```diff
@@ -3466,9 +3519,18 @@ Studio.displayVictoryText = function() {
   var victoryText = document.getElementById('victoryText');
   victoryText.textContent = Studio.victoryText;
   victoryText.setAttribute('visibility', 'visible');
-  var resetText = document.getElementById('resetText');
-  resetText.textContent = studioMsg.tapOrClickToReset();
-  resetText.setAttribute('visibility', 'visible');
+  if (dom.isMobile()) {
```
Comment (davidsbailey) [inline apps/src/studio/studio.js]: should this condition have `dom.isWindowsTouch()` too?

File: apps/src/studio/studio.js (hunk 3 above)
Comment (davidsbailey) [inline apps/src/studio/studio.js:2149]: @caleybrock or @breville are there any other places we should check to make sure this overlap doesn't show up?

Comment (caleybrock) [inline apps/src/studio/studio.js:2149]: not that I know of, testing playlab should cover gumball/iceage/infinity

Comment (aoby) [inline apps/src/studio/studio.js:2149]: I verified it does not show up in those puzzles (in mobile)

File: apps/src/studio/studio.js (hunk 4 above)
Comment (aoby) [inline apps/src/studio/studio.js]: Yes, thanks. updated.

Comment (davidsbailey) [inline apps/src/studio/studio.js]: LGTM

File: apps/src/studio/studio.js (hunk 3 above)
Comment (breville) [inline apps/src/studio/studio.js:2149]: Studio is used for /infinity, /starwars (obviously) and /starwarsblocks, in addition to the ones Caley listed.

**Comment by davidsbailey** [comment]
Andrew, from reading the code it seems like this will show up on any playlab share links. Please test whether it shows up on share levels in other flavors of playlab, e.g.:
1. open /s/playlab/stage/1/puzzle/10 in iOS (or Android)
2. add some blocks, click run, click finish
3. open the share link in a new tab (still in iOS / Android)

I think we may want to gate this overlay on a flag specified in the level definition, so that we can choose exactly which levels to show it on. For now we could just specify that flag on the specific star wars levels we need it on. @joshlory @brendan does this sound right?

**Comment by breville** [comment]
This seems reasonable if we want to reduce our test matrix to some known levels.  I can show which starwars levels would need it.

**Comment by davidsbailey** [comment]
OK, I see what's going on here -- the overlay is already conditioned on `level.tapSvgToRunAndReset` which is only set on hoc2105 level `js_hoc2015_event_free`. So, this PR is already restricted to the last level of the star wars script.

The same flag also adds the actual ability to start the level by tapping the screen. So, we can just add `tapSvgToRunAndReset` to other levels to get the same behavior and overlay.

**Comment by aoby** [comment]
Talked offline w @davidsbailey; merging now and we'll discuss extending to additional levels later.

**Pull request merged**